Extend bucket notification timeout and map to the test timeout

williamlardier opened the pull request:
- Increase timeout for bucket notification tests, to match the current config (ensure the check timeouts before the step)
- Fixes an issue with kafka cleaner, when an error happens, we have no log and we clear the custer timeout, while the step timeout is -1, resulting in builds never completing (6h+), and improve the timeout.
- Add a custom timeout when checking the DMF volume, as the default 100s is too low, and we have this issue very frequently since months.

Seragonia committed:
Extend bucket notification timeout and map to the test timeout

Issue: ZENKO-4806

bert-e commented:
# Hello williamlardier,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/zenko/doc/user).





<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4806 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.10.0`

* `2.6.66`

* `2.7.62`

* `2.8.42`

* `2.9.17`


Please check the `Fix Version/s` of ZENKO-4806, or the target
branch of this pull request.

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4806 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.10.0`

* `2.6.67`

* `2.7.63`

* `2.8.43`

* `2.9.18`


Please check the `Fix Version/s` of ZENKO-4806, or the target
branch of this pull request.

bert-e commented:
# Request integration branches



Waiting for integration branch creation to be requested by the user.

To request integration branches, please comment on this pull request with the following command:

```
/create_integration_branches
```

Alternatively, the `/approve` and `/create_pull_requests` commands will automatically
create the integration branches.

ghost commented:
/create_integration_branches

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/ZENKO-4806` into
`development/2.6`
* [w/2.7/bugfix/ZENKO-4806](https://github.com/scality/zenko/compare/development/2.7...w/2.7/bugfix/ZENKO-4806) will be merged into `development/2.7` 
* [w/2.8/bugfix/ZENKO-4806](https://github.com/scality/zenko/compare/development/2.8...w/2.8/bugfix/ZENKO-4806) will be merged into `development/2.8` 
* [w/2.9/bugfix/ZENKO-4806](https://github.com/scality/zenko/compare/development/2.9...w/2.9/bugfix/ZENKO-4806) will be merged into `development/2.9` 
* [w/2.10/bugfix/ZENKO-4806](https://github.com/scality/zenko/compare/development/2.10...w/2.10/bugfix/ZENKO-4806) will be merged into `development/2.10` 



The following branches will **NOT** be impacted:

* `development/2.5`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **create_integration_branches**

Seragonia committed:
Ensure the KafkaCleaner test completes

An issue where the kafkacleaner never completed was observed in
the CI; as a result, the build took 6h before being cancelled.
THis fix ensures that we tolerate up to 6min.
Note that the current value is 30s * (1000 + 5000) so 3 minutes.

Issue: ZENKO-4806

Seragonia committed:
Extend the duration checks for the DMF volume

In some cases, the platform will take some time to process the
event, so the test will fail, although there is no error.
This change also add a sleep between two checks, to reduce the
impact on the CPU when multiple tests are running this step in
parallel.

Issue: ZENKO-4806

Seragonia committed:
Lint code

Issue: ZENKO-4806

francoisferrand commented on tests/ctst/common/common.ts:
```diff
@@ -237,14 +237,15 @@ Then('i {string} be able to add user metadata to object {string}',
     });
 
-Then('kafka consumed messages should not take too much place on disk', { timeout: -1 },
+const kafkaCleanerTimeout = 1800000; // 30min
```
30 minutes seems verrrry long for kafka cleaner, which is setup to run every 30 second (or maybe 1-2 minutes) in CI...

williamlardier replied:
It's already 30min today, this was the config where we got the best results

williamlardier replied:
I am trying this now https://github.com/scality/Zenko/pull/2135/commits/0125893e807c72653c7605fa4f0f67178692324f

francoisferrand commented on tests/ctst/common/common.ts:
```diff
@@ -307,6 +308,9 @@ Then('kafka consumed messages should not take too much place on disk', { timeout
 
             // If a topic remains in this array, it means it has not been cleaned
             assert(topics.length === 0, `Topics ${topics.join(', ')} still have not been cleaned`);
+        } catch (err: unknown) {
```
is this not done in cucumber/ctst?
should it not be?

williamlardier replied:
We have a finally that clears the timeout even if we got an error, which mean we never stop the test as its timeout was -1

williamlardier replied:
Example: https://github.com/scality/Zenko/actions/runs/10668356308

francoisferrand replied:
what I mean is that (even with the catch) the exception should be propagated: and eventually reach cucumber/ctst where it shoudl be caught.... and immediately fail the test (even with no timeout)

williamlardier replied:
Maybe the timeout was not due to an error... At least this ensures we log the error with a message

Errors not catched by the tests are sent to cucumber directly, it'll mark the test as failed. But in the build above, it didn't work

Seragonia committed:
Increase frequency and checks of kafkacleaner

Issue: ZENKO-4806

ghost commented:
/force_reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **create_integration_branches**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/ZENKO-4806` into
`development/2.6`
* [w/2.7/bugfix/ZENKO-4806](https://github.com/scality/zenko/compare/development/2.7...w/2.7/bugfix/ZENKO-4806) will be merged into `development/2.7` 
* [w/2.8/bugfix/ZENKO-4806](https://github.com/scality/zenko/compare/development/2.8...w/2.8/bugfix/ZENKO-4806) will be merged into `development/2.8` 
* [w/2.9/bugfix/ZENKO-4806](https://github.com/scality/zenko/compare/development/2.9...w/2.9/bugfix/ZENKO-4806) will be merged into `development/2.9` 
* [w/2.10/bugfix/ZENKO-4806](https://github.com/scality/zenko/compare/development/2.10...w/2.10/bugfix/ZENKO-4806) will be merged into `development/2.10` 



The following branches will **NOT** be impacted:

* `development/2.5`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **create_integration_branches**

ghost commented:
Last try to reproduce the errors with fresh branches, if I cannot, I will close this PR, as the current changes are now already part of other PRs
